src/config/database_pools.rs

```diff
@@ -20,15 +24,30 @@ pub struct DatabasePools {
 pub struct DbPoolConfig {
     pub url: String,
     pub read_only_mode: bool,
+    pub pool_size: u32,
+    pub min_idle: Option<u32>,
 }
 
 impl DatabasePools {
     pub fn are_all_read_only(&self) -> bool {
         self.primary.read_only_mode
     }
+
+    pub fn primary_max_pool_size(&self) -> u32 {
+        self.primary.pool_size
+    }
```

Comment:
I'm not sure if it's worth having a function just for this, the rest of the code could access the pool size directly.

Reply:
Removed this to be consistent.